pkg/utils/utils_virtual.go

```diff
@@ -93,55 +93,44 @@ type OSPNetworkData struct {
 	// Omit Services
 }
 
-func metaData(platformType PlatformType, address string) (netFilter string, macAddress string) {
-	switch platformType {
-	case VirtualOpenStack:
-		metaData, networkData := readOpenstackMetaData()
-		netFilter, macAddress = parseOpenstackMetaData(address, metaData, networkData)
-
-	default:
-		glog.V(2).Infof("Unknown PlatformType: %v", platformType)
-	}
-
-	return
-}
-
-func readOpenstackMetaData() (metaData *OSPMetaData, networkData *OSPNetworkData) {
-	networkData = &OSPNetworkData{}
-
-	rawBytes, err := ioutil.ReadFile(ospNetworkData)
+// ReadOpenstackMetaData reads the meta data from the openstack metadata file
+func ReadOpenstackMetaData() (metaData OSPMetaData, err error) {
+	var f *os.File
+	f, err = os.Open(ospMetaData)
 	if err != nil {
-		glog.Errorf("error reading file %s, %v", ospNetworkData, err)
-		return
+		err = fmt.Errorf("error opening file %s: %w", ospMetaData, err)
 	}
-
-	if err = json.Unmarshal(rawBytes, networkData); err != nil {
-		glog.Errorf("error unmarshalling raw bytes %v from %s", err, ospNetworkData)
-		return
+	defer func() {
+		if e := f.Close(); err == nil && e != nil {
+			err = fmt.Errorf("error closing file %s: %w", ospMetaData, e)
+		}
+	}()
+	if err = json.NewDecoder(f).Decode(&metaData); err != nil {
+		err = fmt.Errorf("error unmarshalling metadata from file %s: %w", ospMetaData, err)
 	}
+	return metaData, err
+}
 
-	metaData = &OSPMetaData{}
-
-	rawBytes, err = ioutil.ReadFile(ospMetaData)
+// ReadOpenstackNetworkData reads the network data from the openstack metadata file
+func ReadOpenstackNetworkData() (networkData OSPNetworkData, err error) {
```

Comment:
can we add a glog.info with the name of the functions like we have in other functions